### ISSUE TITLE
Inefficiency in creating P-values

### ISSUE THREAD

**Issue by TomMaullin** [TomMaullin/BLM#76]
The current code for creating P-values often rounds to zero or one. This is not ideal and should be improved upon.

**Comment by nicholst** [comment]
Need to *separately* compute CDF for positive and negative values, and then _only_ use CDF to compute lower tail values... of course... only works for T :(

**Comment by nicholst** [comment]
Best plan:
Catch the warning;
check output for Inf's/NaN and change to a  practical max value, some constant.

**Comment by nicholst** [comment]
FYI... I don't think Python/Numpy/Scipy has any particular problems... checkout these Matlab/Python/R comparisons...  

For a T with DF=10, the CDF 'overflows' to one (p-value underflow) at T=105 in Python, which is worse than Matlab but *better* than R!

![Screenshot 2019-03-19 at 14 40 34](https://user-images.githubusercontent.com/5155907/54616238-4adb3180-4a57-11e9-8916-6584df464b66.png)

Of course, you shouldn't be using the upper tail when you have a T, and for the lower tail all 3 s/w breaks down at about the same point...

![Screenshot 2019-03-19 at 14 56 06](https://user-images.githubusercontent.com/5155907/54616250-50387c00-4a57-11e9-8534-47be60e40cc4.png)

AND!  For the F test, Scipy has behavior different from *all* the others... with it's CDF value 'saturating' earlier than the others, but it plateaus never reaching 1.

![Screenshot 2019-03-19 at 15 06 44](https://user-images.githubusercontent.com/5155907/54617236-d0131600-4a58-11e9-9fd8-652a5abdc0a2.png)


_ANYWAY_ the point is there is no particular horrific numerical performance to blame here...  if you compute a P-value it is alway possible it will underflow to 0 and you need to have an arbitrary "MAX_LOG_PVALUE" that you fill in when you would otherwise attempt to compute -log10(0).

**Comment by TomMaullin** [comment]
Hmm okay... I will check if there is any particular numerical casting happening during the code that could be leading to `-inf`'s and let you know how that goes.

**Comment by nicholst** [comment]
There is no mystery about -inf’s… log(0) is -inf… right?

**Comment by TomMaullin** [comment]
Apologies I should have been clearer. I meant perhaps the underflow to zero is occuring more often than expected and perhaps this could be caused, instead of by the `scipy` stats tools, by other numpy methods casting the matrices to lower precision during some previous operations.

**Comment by nicholst** [comment]
Ah... casting issues... right.

**Comment by TomMaullin** [comment]
Hi @nicholst ,

I have written some python code to check for the minimum log10(1-p) value as shown below:

```
import nibabel as nib
import numpy as np
from scipy import stats

xlist = np.arange(90, 500, 0.1)
dflist = np.arange(2,500)

minval_p = 0;
for df in dflist:
    for x in xlist:
        
        p = np.log10(stats.t.cdf(-x, df))

        if not np.isinf(p):
            
            minval_p = np.minimum(minval_p, p)
            
    print(minval_p)
```

This seems to cap at a value of `-323.3062153431158`. If you are happy with this shall I go ahead and hard code a replacement for `-inf` with this value?

**Comment by nicholst** [comment]
Sure, call it `-323.3062153431158`, but make it a well-documented, easily changed constant.  Realise that this could vary by platform, but I'm not bothered by that.

**Comment by TomMaullin** [comment]
This issue has now been resolved in PR #80
